backend/src/v1/data/repositories/wordRepository.ts

```diff
@@ -1,80 +1,88 @@
-            .where("word.word_id", "=", id)
-            .executeTakeFirstOrThrow();
+type GetWords = (languageCode?: string) => Promise<Word[]>;
+type GetWord = (id: number) => Promise<Word>;
+type InsertWord = (data: InsertData) => Promise<{ insertedRows: string }>;
+type UpdateWord = (data: UpdateData) => Promise<{ updatedRows: string }>;
+type DeleteWord = (id: number) => Promise<{ deletedRowsCount: string }>;
+
+export const getWords: GetWords = async (languageCode) => {
+	const query = db
+		.selectFrom('word')
+		.selectAll()
+		.$if(languageCode !== undefined, (builder) =>
+			builder.where('word.language_code', '=', languageCode!),
+		);
+
+	try {
+		const result: Word[] = await query.execute();
+		return result;
+	} catch (error) {
+		if (typeof error === 'string') {
+			throw new Error(error);
+		} else if (error instanceof Error) {
+			throw new Error(error.message);
+		}
+		return [];
+	}
+};
 
-        return {
-            id: data?.word_id,
-            text: data?.word_text,
-            languageCode: data?.language_code
-        };
-    },
-    insertWord: async (data: WordInsert): Promise<{ insertedRows: string; }> => {
-        let query = db.insertInto("word").values({
-            word_text: data.text,
-            language_code: data.languageCode
-        });
-        let result: InsertResult = await query.executeTakeFirstOrThrow();
-        return { insertedRows: result.numInsertedOrUpdatedRows?.toString() ?? "0" };
-    },
-    updateWord: async (data: WordUpdate): Promise<{ updatedRows: string; }> => {
-        let query = db.updateTable("word")
-            .set({
-                word_text: data.text,
-                language_code: data.languageCode
-            })
-            .where('word_id', '=', data.id);
-        db.updateTable;
-        let result = await query.executeTakeFirstOrThrow();
-        return { updatedRows: result.numUpdatedRows.toString() };
-    },
-    deleteWord: async (id: number): Promise<{ deletedRowsCount: string; }> => {
-        let query = db.deleteFrom("word").where("word.word_id", "=", id);
-        let result: DeleteResult = await query.executeTakeFirstOrThrow();
-        return {
-            deletedRowsCount: result.numDeletedRows.toString()
-        };
-    }
+export const getWord: GetWord = async (id) => {
+	const data = await db
+		.selectFrom('word')
+		.selectAll()
+		.where('word.word_id', '=', id)
+		.executeTakeFirst();
+	if (!data) {
+		throw new Error(`Word with id ${id} not found`);
+	}
+	return data;
 };
 
-export function ProvideWordRepository(): WordRepository {
-    return WordRepositoryImpl;
+export const insertWord: InsertWord = async (
+	data: InsertData,
+): Promise<{ insertedRows: string }> => {
+	const query = db.insertInto('word').values({
+		word_text: data.text,
+		language_code: data.languageCode,
+	});
+	const result = await query.executeTakeFirstOrThrow();
+	return { insertedRows: result.numInsertedOrUpdatedRows?.toString() ?? '0' };
```

Comment:
It's usually nicer to return the newly-inserted row from an insert function. That way you can access its ID if it's autogenerated, for example.